BasicObstacleAvoidence/BasicObstacleAvoidence.ino

```diff
@@ -31,23 +32,39 @@ void setup() {
   });
 
   car.enableCruiseControl();
-  car.setSpeed(1.5); // Maintain a speed of 1.5 m/sec
+ // car.setSpeed(1.5);
 }
 
 void loop() {
   // Maintain the speed and update the heading
-  car.update();
-  int colli = front.getDistance();        //distance measured by the ultrasonic sensors
-  Serial.println("obstacle at  : ", colli);
-
-  car.getHeading();
-  car.getDistance();
+    car.update();
+   
+    while (Serial.available()) blue = Serial.read(); 
+    
+    //Serial.println("obstacle at  : "+ colli);
+  
   
-    if (colli<20 && colli>0){
-      car.setSpeed(0);        //stop moving car
-    }  
-    else{
-      car.setSpeed(1.5);
-    }
+      if(blue == '1'){            //Checks whether value of Incoming_value is equal to 1 
+
+          int colli = front.getDistance();        //distance measured by the ultrasonic sensors
+
+          if (colli<30 && colli>0){
+            car.setSpeed(0);        //stop moving car
+            
+          }  
+          else{
+            car.setSpeed(1.5);
+            car.getHeading();
+            car.getDistance();
+            
+          }
+          
+      }
+      
+      else if(blue == '0'){       //Checks whether value of Incoming_value is equal to 0
+          
+          car.setSpeed(0);        //stop moving car
+      }
   
+    
 }
```

Comment:
Why do you have two Arduino sketches?

Reply:
We will be making changes to this soon and commit just one file. Wanted to have two different files one for Bluetooth and one for obstacle detection sketch but i see there is no reason for that either so will be fixing it very soon with the next commit. :)